microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/model/Property.java

```diff
@@ -129,11 +134,15 @@ public String getPropertyName() {
 	 *         otherwise.
 	 */
 	public String getPropertyNameWithProfile() {
+		return getPropertyNameWithProfile(false);
+	}
+
+	public String getPropertyNameWithProfile(boolean skipMultiLine) {

```

Comment:
Do we need this method signature ?

Reply:
Having this method signature helps because without it, we get this case:
![image](https://user-images.githubusercontent.com/20326645/77008118-f0abcf80-693b-11ea-9d5e-dff759ab7cae.png)
Even though `%prod.quarkus.application.name` exists (as a multiline property key), the completion options provide it in this case when it shouldn't

If these new method signatures are OK, I will go ahead and change the javadoc if you wish

Reply:
@xorye I understand that getPropertyNameWithProfile must return the property name by skipping multi line to retrieve it in hover, validation, definition, etc and I think we should have just one method getPropertyNameWithProfile () which skip multi line and that's it (you can keep tour code with getPropertyNameWithProfile(boolean skipMultiLine) like now but please set this method to private. The idea is to provide a clean API for user which wishes consumes the PropertiesModel.

Reply:
> If these new method signatures are OK, I will go ahead and change the javadoc if you wish

If you can remove it (to avoid having new methods), it should be better and getPropertyNameWithProfile , getPropertyName etc must skip multi line. It should be very good if you could clarify that in the javadoc of each methods.